src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java

```diff
@@ -26,39 +27,42 @@
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("api/categories")
-public class FinancialTransactionCategoryController{
+public class FinancialTransactionCategoryController {
+
+    private final FinancialTransactionCategoryService financialTransactionCategoryService;
 
     @GetMapping("/{id}")
-    ResponseEntity<FinancialTransactionCategoryDetailedDTO> getFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable Long id){
+    ResponseEntity<FinancialTransactionCategoryDetailedDTO> getFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable Long id) {
         //TODO Necessary code implementation
         FinancialTransactionCategoryDetailedDTO financialTransactionCategoryDetailedDTO = null;
         return new ResponseEntity<>(financialTransactionCategoryDetailedDTO, HttpStatus.OK);
     }
 
     @GetMapping()
-    ResponseEntity<List<FinancialTransactionCategoryDTO>> getFinancialTransactionCategories(){
+    ResponseEntity<List<FinancialTransactionCategoryDTO>> getFinancialTransactionCategories() {
         //TODO Necessary code implementation
         List<FinancialTransactionCategoryDTO> financialTransactionCategoryDTOList = null;
         return new ResponseEntity<>(financialTransactionCategoryDTOList, HttpStatus.OK);
     }
 
     @PostMapping()
-    public ResponseEntity<FinancialTransactionCategoryDTO> createFinancialTransactionCategory(@Valid @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO){
-        //TODO Necessary code implementation
-        FinancialTransactionCategoryDTO financialTransactionCategoryDTO = null;
+    public ResponseEntity<FinancialTransactionCategoryDTO> createFinancialTransactionCategory(
+            @Valid @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO) {
```

Comment:
When you are using annotation `@Valid` then `@Validated` on the class is necessary as well.
Anyway please remove `@Valid` from this method and add it to the service method instead. As well please add `@Validated` to the service class.